src/lang-promql/client/prometheus.ts

```diff
@@ -101,18 +103,27 @@ export class HTTPPrometheusClient implements PrometheusClient {
     if (config.fetchFn) {
       this.fetchFn = config.fetchFn;
     }
+    if (config.httpMethodUsed) {
+      this.httpMethodUsed = config.httpMethodUsed;
+    }
   }
 
   labelNames(metricName?: string): Promise<string[]> {
     const end = new Date();
     const start = new Date(end.getTime() - this.lookbackInterval);
     if (metricName === undefined || metricName === '') {
-      const params: URLSearchParams = new URLSearchParams({
-        start: start.toISOString(),
-        end: end.toISOString(),
-      });
+      const params = this.buildURIAndBody(
```

Comment:
At this point, `params` does not seem like the right name anymore, since it stores more than just the URL params. How about `request` or something like that, in which case `buildURIAndBody()` could be called `buildRequest()` as well?

Reply:
I didn't feel confortable with this name anyway ^^. `request` + `buildRequest()` are fine for me